catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/commands/SearchFormsLoaderCommand.java

```diff
@@ -0,0 +1,123 @@
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.commands;
+
+import static org.codice.ddf.catalog.ui.forms.SearchFormsUtil.allNull;
+import static org.fusesource.jansi.Ansi.ansi;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import java.io.PrintStream;
+import java.util.List;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.codice.ddf.catalog.ui.forms.SearchFormsLoader;
+import org.codice.ddf.catalog.ui.forms.SearchFormsUtil;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.codice.ddf.commands.catalog.SubjectCommands;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(
+  scope = "forms",
+  name = "load",
+  description = "Handles the ingestion of search forms and result set forms into the system"
+)
+public class SearchFormsLoaderCommand extends SubjectCommands {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SearchFormsLoaderCommand.class);
+
+  @Reference CatalogFramework catalogFramework;
+
+  @Reference EndpointUtil endpointUtil;
+
+  @Option(
+    name = "--formsDirectory",
+    required = false,
+    aliases = {"-d"},
+    multiValued = false,
+    description = "Overrides the default forms directory path. Default: /etc/forms"
+  )
+  protected String formsDir = null;
+
+  @Option(
+    name = "--forms",
+    required = false,
+    aliases = {"-f"},
+    multiValued = false,
+    description = "Override the default forms.json file name. Default: forms.json"
+  )
+  protected String formsFile = null;
+
+  @Option(
+    name = "--results",
+    required = false,
+    aliases = {"-r"},
+    multiValued = false,
+    description = "Override the default results.json file name. Default: results.json"
+  )
+  protected String resultsFile = null;
+
+  @Override
+  protected Object executeWithSubject() {
+    final PrintStream console = System.out;
+    final SearchFormsLoader loader;
+
+    console.println(ansi().fgBrightCyan().a("Initializing Search Form Template Loader").reset());
+
+    if (SearchFormsUtil.allNull(formsDir, formsFile, resultsFile)) {
+      console.println(
+          ansi()
+              .fgYellow()
+              .a("No options selected, falling back to default directory locations.")
+              .reset());
+      loader = generateLoader(catalogFramework, endpointUtil, null, null, null);
```

Comment:
Would it be better for you to have 2 constructors, instead of one that you're passing 3 null parameters?

Reply:
Not in my opinion, don't see much benefit.